### PR TITLE
Added support for '-'-separated time in FITS (`utils.get_obs_time`)

### PR DESCRIPTION
**Pull request by vlad2305m**

### CODE REVIEW

**Comment by karpov-sv** [comment]
The code will actually need to also handle the difference of keywords that TAROT uses. Like that:
```diff
--- a/stdpipe/utils.py
+++ b/stdpipe/utils.py
@@ -164,7 +164,7 @@ def get_obs_time(
         log('Loading FITS header from', filename)
         header = fits.getheader(filename)
 
-    for dkey in ['DATE-OBS', 'DATE', 'TIME-OBS', 'UT', 'MJD', 'JD']:
+    for dkey in ['DATE-OBS', 'DATEOBS', 'DATE', 'TIME-OBS', 'TIMEOBS', 'UT', 'MJD', 'JD']:
         if dkey in header:
             log('Found ' + dkey + ':', header[dkey])
             # First try to parse standard ISO time
@@ -173,7 +173,7 @@ def get_obs_time(
             except:
                 log('Could not parse ' + dkey + ' using Astropy parser')
 
-            for tkey in ['TIME-OBS', 'UT']:
+            for tkey in ['TIME-OBS', 'TIMEOBS', 'UT']:
                 if tkey in header:
                     log('Found ' + tkey + ':', header[tkey])
                     try:
```

**Review by karpov-sv — CHANGES_REQUESTED** [review]

File: stdpipe/photometry.py
[the remark this answers is not in the document]
Comment (vlad2305m) [reply stdpipe/photometry.py:1]:
```
  File "./stdpipe/photometry.py", line 1091, in measure_objects
    bg_est = photutils.background.Background2D(
             ^^^^^^^^^^^^^^^^^^^^
AttributeError: module 'photutils' has no attribute 'background'
```
For me photometry doesn't work without those

File: stdpipe/utils.py [hunk 1]
```diff
@@ -144,6 +144,11 @@ def convert_time(time):
                 time = Time(time, format='unix')
 
         else:
+            if isinstance(time, str):
+                # Special case where some telescope used '-' for time separation instead of ISO ':'
+                m=re.fullmatch(r"(\d\d\d\d-\d\d-\d\dT\d\d)-(\d\d)-(\d\d\.\d\d\d)", time.strip())
```
[the remark this answers is not in the document]
Comment (vlad2305m) [reply stdpipe/utils.py]: Done, tested. Astropy adds `.000` at the end. Is that a problem?

[the remark this answers is not in the document]
Comment (karpov-sv) [reply stdpipe/utils.py]: It is buggy:
```python
re.fullmatch(r"(\d\d\d\d-\d\d-\d\dT\d\d)-(\d\d)-(\d\d\.?\d?\d?\d?)", "2024-12-01T12-22-33123")
Out[12]: <re.Match object; span=(0, 22), match='2024-12-01T12-22-33123'>
```
Please use the form I suggested above (with mandatory dot before fractional part), maybe modifying it to accept 1 to 6 digits - like `(\d\d(\.\d{1,6})?)`

File: stdpipe/photometry.py
[the remark this answers is not in the document]
Comment (karpov-sv) [reply stdpipe/photometry.py:1]: Oh, I hate it. It works in 1.9.0 and indeed does not work in 2.0.2... 
Ok, let it be so.

File: stdpipe/utils.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (vlad2305m) [reply stdpipe/utils.py]: Done. The tricky part is not capturing it as a group

[the remark this answers is not in the document]
Comment (karpov-sv) [reply stdpipe/utils.py]: You might as well just use `":".join(m.groups()[:3])` later, but your approach also works.
I will merge and try it on production now.

**Comment by karpov-sv** [comment]
Seems working as expected :-)
http://grandma-stdpipe.ijclab.in2p3.fr/tasks/3667

**Pull request merged**